src/network/network_content.cpp

```diff
@@ -804,7 +815,32 @@ void ClientNetworkContentSocketHandler::DownloadContentInfo(ContentID cid)
 	if (std::ranges::find(this->requested, cid) != this->requested.end()) return;
 
 	this->requested.push_back(cid);
-	this->RequestContentList(1, &cid);
+	this->queued.push_back(cid);
+	_request_queue_timeout.Reset();
+}
+
+/**
+ * Send a content request for queued content info download.
+ */
+void ClientNetworkContentSocketHandler::RequestQueuedContentInfo()
+{
+	if (this->queued.empty()) return;
+
+	/* Wait til we've briefly stopped receiving data (which will contain more content) before making the request. */
+	if (std::chrono::steady_clock::now() <= this->last_activity + CONTENT_QUEUE_TIMEOUT) {
+		_request_queue_timeout.Reset();
+		return;
+	}
+
+	/* We're doing it. Move the queue locally so more ids can be queued for later. */
+	ContentIDList queue = std::move(this->queued);
+
+	/* Remove ids that have since been received since the request was queued. */
+	queue.erase(std::remove_if(std::begin(queue), std::end(queue), [this](ContentID content_id) {
+		return std::ranges::find(this->infos, content_id, &ContentInfo::id) != std::end(this->infos);
+	}), std::end(queue));
```

Comment:
There is nothing to prevent the system from adding the same not-yet-received ID into the queue multiple times, right? Should that be prevented, or is that too insignificant to worry about?